projects/ion/src/lib/sidebar/sidebar.component.ts

```diff
@@ -14,8 +14,35 @@ export class IonSidebarComponent {
 
   public closed = true;
 
+  public checkClikOnPageAccess = (
+    event
+  ): EventListenerOrEventListenerObject => {
+    this.checkClikOnPage(event);
+    return;
+  };
+
+  public checkClikOnPage(event): EventListenerOrEventListenerObject {
+    const containerElement = [document.querySelector('.ion-sidebar--opened')];
+    const innerElement = event.target;
+    if (containerElement.length && !containerElement.includes(innerElement)) {
+      const closeButton = document.querySelector(
+        '.ion-sidebar--opened .ion-sidebar__header button'
+      ) as HTMLElement;
+      if (closeButton) {
+        closeButton.click();
+      }
+      document.removeEventListener('click', this.checkClikOnPage);
+    }
+    return;
+  }
+
   public toggleVisibility(): void {
     this.closed = !this.closed;
+    if (!this.closed) {
+      setTimeout(() => {
+        document.addEventListener('click', this.checkClikOnPageAccess);
```

Comment:
Acredito que você poderia chamar diretamente `checkClikOnPage`;

```suggestion
        document.addEventListener('click', this.checkClikOnPage);
```

Reply:
Se eu fizer isso, ele quebra no teste. Aparentemente é devido o tempo de um querySlector da função capturar o elemento na tela. O addEventListener, nesse caso em si, não pode chamar direto uma função, por conta que preciso do parâmtro implícito que ele passa, e a forma mais prática para isso é se essa função estiver contida numa variável, ao invés de chamar a função propriamente dita.

Então, em resumo: tem a variável com função por conta do addEventListener, e tem a função separado por conta do teste.